### PR TITLE
Adding cumulative_trapezoid function

### LINKED ISSUE #12708: cumulative_trapezoid

### PR DESCRIPTION
**Pull request by jatinS-dev**
closes #12708

### CODE REVIEW

File: ivy/functional/frontends/torch/blas_and_lapack_ops.py [hunk 1]
```diff
@@ -210,3 +210,10 @@ def trapezoid(y, x=None, *, dx=None, dim=-1):
     if x is not None:
         y, x = torch_frontend.promote_types_of_torch_inputs(y, x)
     return ivy.trapz(y, x=x, dx=dx, axis=dim)
+
+
+@to_ivy_arrays_and_back
+def cumulative_trapezoid(y, x=None, *, dx=None, dim=-1):
+    if x is not None:
+        y, x = torch_frontend.promote_types_of_torch_inputs(y, x)
+    return ivy.cumulative_trapezoid(y, x=x, dx=dx, axis=dim)
```
Comment (Daniel4078) [inline ivy/functional/frontends/torch/blas_and_lapack_ops.py:219]: there is no cumulative_trapezoid function in ivy for you to use directly like this, please consider compositional approach or something else.

**Review by Daniel4078 — CHANGES_REQUESTED** [review]
Hello, I found that your implementation for the frontend does not work. Please check the comment below, thanks!

File: ivy/functional/frontends/torch/blas_and_lapack_ops.py (hunk 1 above)
Comment (jatinS-dev) [reply ivy/functional/frontends/torch/blas_and_lapack_ops.py:219]: Hello @Daniel4078 so should I make another PR for cumulative_trapezoid or continue the function development here only, by using the composition function?

Comment (Daniel4078) [reply ivy/functional/frontends/torch/blas_and_lapack_ops.py:219]: you should just continue here first as there might be some functions in the Ivy api that provide similar feature like cumulative_trapezoid but named differently.

**Review by Daniel4078 — CHANGES_REQUESTED** [review]
Well, your implementation somehow does not really work. For example, according to the test log here:(https://github.com/unifyai/ivy/actions/runs/4499836402/jobs/7918203393?pr=12778). your function produce the wrong result "[2. 4.]" when the expected result is "[[2. 2.]] " for inputs: dtype_y_x=(['float32'], array([[2., 2.],
E               [2., 2.]], dtype=float32), array([[2., 2.],
E               [2., 2.]], dtype=float32)),
E       use_x=False,
E       dim=0,
E       dx=1.0,

**Comment by jatinS-dev** [comment]
@Daniel4078 how can I check this error on my local machine?

**Comment by Daniel4078** [comment]
> @Daniel4078 how can I check this error on my local machine?

I think you should check the deep dive document about how to run certain tests on your local machines.

**Pull request closed without merging**